internal/controller/apiextensions/claim/api.go

```diff
@@ -63,8 +63,14 @@ func (a *APIBinder) Bind(ctx context.Context, cm resource.CompositeClaim, cp res
 		return errors.New(errBindClaimConflict)
 	}
 
-	// Propagate the actual external name back from the composite to the claim.
-	meta.SetExternalName(cm, meta.GetExternalName(cp))
+	// Propagate the actual external name back from the composite to the claim
+	// if it's set.
+	// For dynamically provisioned composites, claim's external name is the
+	// source initially (if set) as expected.
+	en := meta.GetExternalName(cp)
+	if en != "" {
```

Comment:
```suggestion
	if en := meta.GetExternalName(cp); en != "" {
```

I just noticed we could do this to reduce the scope of the `en` variable. Just a nit pick though - feel free to ignore.